documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc

```diff
@@ -1022,13 +1025,33 @@ Next, functions for working with numeric values:
 | `exp()` | Exponential function | `exp(x)` | Universal in SQL dialects
 | `power()` | Exponentiation | `power(x,y)` | Universal in SQL dialects
 | `ln()` | Natural logarithm | `ln(x)` | Very common in SQL dialects
+| `log()` | Base-N logarithm | `log(x,x)` | Very common in SQL dialects
+| `log2()` | Base-2 logarithm | `log2(x)` | Very common in SQL dialects
```

Comment:
Rollback please.

Reply:
1. `log2()` is only supported on MySQL. It definitely doesn't belong in the list of standard HQL functions. I mean, we _could_ emulate it using `ln(x)/ln(2)` but that is not currently implemented. I don't think we need it.
2. `log(b,x)` _is_ supported natively on _most_ of the built-in dialects, but not all, and so if you want to add it to the list, you need to add code to emulate it using `ln(x)/ln(b)`.